app/components/primer/avatar_component.rb

```diff
@@ -34,17 +34,25 @@ def initialize(src:, alt:, size: 20, square: false, href: nil, **system_argument
       @system_arguments[:height] = size
       @system_arguments[:width] = size
 
-      @system_arguments[:classes] = class_names(
-        system_arguments[:classes],
-        "avatar" => !href,
+      avatar_class_names = class_names(
+        "avatar",
         "avatar--small" => size < SMALL_THRESHOLD,
         "circle" => !square
       )
+
+      if @href
+        @link_arguments = { href: href, classes: avatar_class_names }
+      else
+        @system_arguments[:classes] = class_names(
+          system_arguments[:classes],
+          avatar_class_names
+        )
+      end
```

Comment:
```suggestion
      avatar_class_names = class_names(
        "avatar",
        system_arguments[:classes],
        "avatar--small" => size < SMALL_THRESHOLD,
        "circle" => !square
      )

      if @href
        @link_arguments = { href: href, classes: avatar_class_names }
      else
        @system_arguments[:classes] = avatar_class_names
      end
```

this is what I had in mind